### PR TITLE
feat: Implement delete site action menu - EXO-64952 - Meeds-io/MIPs#68

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java [hunk 1]
```diff
@@ -68,4 +70,26 @@ public Response getSites() {
       return Response.serverError().build();
     }
   }
+
+  @DELETE
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @Operation(summary = "removed sites", method = "GET", description = "This removes sites")
```
Comment (azayati) [inline layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java]:
```suggestion
  @Operation(summary = "Delete a site", method = "GET", description = "This deletes the given site")
```

File: layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java [hunk 2]
```diff
@@ -68,4 +70,26 @@ public Response getSites() {
...
+  String siteType,
+                             @Parameter(description = "site name")
+                             @QueryParam("siteName")
+                             String siteName) {
+    try {
+      SiteId siteId = Util.from(new SiteKey(siteType, siteName));
+      portal.removeSite(siteId);
+      return Response.ok().build();
+    } catch (Exception e) {
+      LOG.error("Error when removing sites", e);
```
Comment (azayati) [inline layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java]:
```suggestion
      LOG.error("Error when deleting the site with name {} and type {}", siteName, siteType, e);
```

File: layout-management-webapps/src/main/webapp/vue-app/site-management/siteManagementService.js [hunk 3]
```diff
@@ -30,4 +30,19 @@ export function getSites() {
...
+  const formData = new FormData();
+  formData.append('siteName', siteName);
+  formData.append('siteType', siteType);
+  const params = new URLSearchParams(formData).toString();
+  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/v1/sites?${params}`, {
+    method: 'DELETE',
+    credentials: 'include',
+  }).then(resp => {
+    if (!resp || !resp.ok) {
+      throw new Error('Error when deleting sites');
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-management/siteManagementService.js]:
```suggestion
      throw new Error('Error when deleting site');
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java [hunk 4]
```diff
@@ -20,14 +20,16 @@
 import java.util.List;
 
 import javax.annotation.security.RolesAllowed;
-import javax.ws.rs.GET;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
+import javax.ws.rs.*;
```
Comment (azayati) [inline layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java]:
```suggestion
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java (hunk 1 above)
Comment (azayati) [reply layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java]: to be considered @GouadriaHanen

**Review by azayati — CHANGES_REQUESTED** [review]

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**